Mathlib/Topology/Algebra/Algebra/Equiv.lean

```diff
@@ -0,0 +1,303 @@
+/-
+Copyright (c) 2024 Salvatore Mercuri. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Salvatore Mercuri
+-/
+import Mathlib.Topology.Algebra.Algebra
+
+/-!
+# Isomorphisms of topological algebras
+
+This file contains an API for `ContinuousAlgEquiv R A B`, the type of
+continuous `R`-algebra isomorphisms with continuous inverses. Here `R` is a
+commutative (semi)ring, and `A` and `B` are `R`-algebras with topologies.
+
+## Main definitions
+
+Let `R` be a commutative semiring and let `A` and `B` be `R`-algebras which
+are also topological spaces.
+
+* `ContinuousAlgEquiv R A B`: the type of continuous `R`-algebra isomorphisms
+  from `A` to `B` with continuous inverses.
+
+## Notations
+
+`A ≃A[R] B` : notation for `ContinuousAlgEquiv R A B`.
+
+## Tags
+
+* continuous, isomorphism, algebra
+-/
+
+open scoped Topology
+
+
+/--
+`ContinuousAlgEquiv R A B`, with notation `A ≃A[R] B`, is the type of bijections
+between the topological `R`-algebras `A` and `B` which are both homeomorphisms
+and `R`-algebra isomorphisms.
+-/
+structure ContinuousAlgEquiv (R A B : Type*) [CommSemiring R]
+    [Semiring A] [TopologicalSpace A] [Semiring B] [TopologicalSpace B] [Algebra R A]
+    [Algebra R B] extends A ≃ₐ[R] B where
+  continuous_toFun : Continuous toFun := by continuity
+  continuous_invFun : Continuous invFun := by continuity
```

Comment:
Is there a reason why this is not simply `extends A ≃ₐ[R] B, A ≃ₜ B`? The fields have the same names.

Reply:
It seems to work (plus the analogous change to the `...Class` below) and has the effect of not needing to separately define `ContinuousAlgEquiv.toHomeomorph`.

Reply:
I see that this now triggers linter warnings. I'm a bit confused about these: why does it want a docstring for an autogenerated instance? And the second one does not say from what the lhs simpifies...
Are these possibly bugs?

Reply:
I've silenced the docBlame linter for the first issue. For the second one, `#lint` in the file reports that the `dsimp` lemma `Homeomorph.homeomorph_mk_coe` changes `⇑e.toHomeomorph` to `⇑e.toEquiv`, then `AlgEquiv.toEquiv_eq_coe` changes it to `⇑↑e.toAlgEquiv` and then finally `EquivLike.coe_coe` changes it to `⇑e.toAlgEquiv`, so I've removed the bad simp lemma and added a new one changing `⇑e.toAlgEquiv` to `⇑e`.